### ISSUE TITLE
Branding does not show up correctly on grubmenu with kairos 2.1.1

### ISSUE THREAD

**Issue by santhoshdaivajna** [kairos-io/kairos#1563]
```
**Kairos version:**
root@edge-89a33842be829ddebe91aba70105bcfc:~# cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.2 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.2 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
KAIROS_NAME="kairos-core-ubuntu-22-lts"
KAIROS_VERSION="v2.1.1"
KAIROS_ID="kairos"
KAIROS_ID_LIKE="kairos-core-ubuntu-22-lts"
KAIROS_VERSION_ID="v2.1.1"
KAIROS_PRETTY_NAME="kairos-core-ubuntu-22-lts v2.1.1"
KAIROS_BUG_REPORT_URL="https://github.com/kairos-io/kairos/issues"
KAIROS_HOME_URL="https://github.com/kairos-io/kairos"
KAIROS_IMAGE_REPO="quay.io/kairos/core-ubuntu-22-lts"
KAIROS_IMAGE_LABEL="latest"
KAIROS_GITHUB_REPO="kairos-io/kairos"
KAIROS_VARIANT="core"
KAIROS_FLAVOR="ubuntu-22-lts"KAIROS_NAME="core-ubuntu-22-lts-k3s:v3.4.7-alpha2"
KAIROS_VERSION="1.24.6-k3s1_v3.4.7-alpha2"
KAIROS_ID="core-ubuntu-22-lts-k3s"
KAIROS_ID_LIKE="core-ubuntu-22-lts-k3s:v3.4.7-alpha2"
KAIROS_VERSION_ID="1.24.6-k3s1_v3.4.7-alpha2"
KAIROS_PRETTY_NAME="core-ubuntu-22-lts-k3s:v3.4.7-alpha2 1.24.6-k3s1_v3.4.7-alpha2"
KAIROS_BUG_REPORT_URL=""
KAIROS_HOME_URL=""
KAIROS_IMAGE_REPO="gcr.io/spectro-dev-public/aslam"
KAIROS_IMAGE_LABEL="v2.1.1_1.24.6-k3s1_v3.4.7-alpha2"
KAIROS_GITHUB_REPO=""
KAIROS_VARIANT="
```

**CPU architecture, OS, and Version:**
`Linux edge-89a33842be829ddebe91aba70105bcfc 5.19.0-45-generic #46~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Jun 7 15:06:04 UTC 20 x86_64 x86_64 x86_64 GNU/Linux`

**Describe the bug**
Branding does not show up fine on the grubmenu. Please find the attached screenshot.
Kairos 2.1.1
![Screenshot 2023-06-27 at 2 30 35 PM (2)](https://github.com/kairos-io/kairos/assets/504862/ecc1e014-1d66-4088-bd8c-22a31a0ba84c)

Kairos 2.2.0
<img width="1382" alt="Screenshot 2023-06-28 at 8 12 28 PM" src="https://github.com/kairos-io/kairos/assets/504862/3f9ebd53-5f4d-4aed-8108-9b81423ba927">

**To Reproduce**

**Expected behavior**
Branding to show up correctly on the grubmenu.

**Logs**

**Additional context**

**Comment by mauromorales** [comment]
@santhoshdaivajna thanks for reporting this. The issue is indeed present in 2.1 but was fixed in 2.2. There are no plans of backporting this fix atm so the best would be to upgrade your karios system.

**Comment by mauromorales** [comment]
We will do a 2.1.2 release for this. The CI takes a bit though, so you can expect it for Monday but I'll ping you as soon as it is ready

**Comment by Itxaka** [comment]
in the first image, the 2 extra options at the bottom are the ones coming from the grubmenu.cfg, which extends the default grub so the grubmenu override works.

What doesnt work is the default grub entry name, which is set via the elemental config. This is a known issue in kairos-agent 2.1.1 which comes from the full rework of the config to merge kairos-agent and elemental cli.

kairos 2.1.3 is the one that fixes this by loading  the elemental config instead of creating a new one.

This can be easily checked by once booting confirming that there is a file in `/run/initramfs/cos-state/grub_oem_env` which contains the `default_menu_entry` key with a value. 

If that file has `cOS` as the value it means that the issue comes from the agent not reading the elemental config files during install.

Our default grub reads several files from the state partition in which we are able to override the default values:

 - grubenv (from the oem partition, not state!)
 - grub_oem_env
 - grubmenu
 - grubcustom


grubmenu is used to EXTEND the grub menu with extra entries, it does not contain any vars
grubcustom is used for the boot assesment, can also extend the grub menu
grubenv is used to set the entries and allow overriding values for users during a running system (`grub2-editenv /oem/grubenv set next_entry=statereset`)
grub_oem_env is ONLY used for the default_entry_name and ONLY created during install. It allows to setup oem values that will be applied during install and could be extended in the future to provide more configurations for the default boot.

it should be impossible to set variables via the grubmenu.cfg as that is a menu entry file not an env file and they have different formats and can only be edited via grub2-editenv.

So the fix would be to update the repos on 2.1.3 to grab the kairos-agent v2.1.3 which should fix this.

**Comment by mudler** [comment]
This should be fixed in https://github.com/kairos-io/kairos/releases/tag/v2.1.3 . Please let us know and re-open if necessary Thanks @mauromorales and @Itxaka !
